lib/Parse/ParseIfConfig.cpp

```diff
@@ -544,8 +547,97 @@ static bool isVersionIfConfigCondition(Expr *Condition) {
   return IsVersionIfConfigCondition().visit(Condition);
 }
 
+/// Get the identifier string from an \c Expr if it's an
+/// \c UnresolvedDeclRefExpr, otherwise the empty string.
+static StringRef getDeclRefStr(Expr *E) {
+  if (auto *UDRE = cast<UnresolvedDeclRefExpr>(E)) {
```

Comment:
Am I just mistaken, or shouldn't this be a `dyn_cast`?  `cast` implies that this must succeed and will abort if the expression type is not a `UnresolvedDeclRefExpr`.

Reply:
Oh, yes, this is a typo, thanks! Will fix tomorrow.